cli/integration_tests/run_summary/strict_env.t

```diff
@@ -0,0 +1,206 @@
+  $ ${TURBO} run build --experimental-env-mode=loose --summarize > /dev/null
+  $ cat .turbo/runs/*.json | jq -r '.envMode'
+  loose
+  $ cat .turbo/runs/*.json | jq -r '.tasks[0].envMode'
+  loose
+  $ cat .turbo/runs/*.json | jq -r '.tasks[0].environmentVariables | {passthrough, globalPassthrough}'
+  {
+    "passthrough": [
+      "LOCAL_VAR_PT=7cd1bb19c058cf4d6ad6aa579d685bddddf3ab587b78bdcb1e6e488fb6f47a3b"
+    ],
+    "globalPassthrough": [
+      "GLOBAL_VAR_PT=cecd31fff1e723588eac8fe1edff89a6d2ec072f5c3bd884a98297487670b1f0"
+    ]
+  }
+
+Global passthrough specified empty array + infer
+  $ rm -rf .turbo/runs
+  $ cp "$TESTDIR/fixture-configs/global_pt-empty.json" "$(pwd)/turbo.json" && git commit --allow-empty -am "no comment" --quiet
+  $ ${TURBO} run build --summarize > /dev/null
+  $ cat .turbo/runs/*.json | jq -r '.envMode'
+  strict
+  $ cat .turbo/runs/*.json | jq -r '.tasks[0].envMode'
+  strict
+  $ cat .turbo/runs/*.json | jq -r '.tasks[0].environmentVariables | {passthrough, globalPassthrough}'
+  {
+    "passthrough": null,
+    "globalPassthrough": []
+  }
+
+Global passthrough specified value + infer
+  $ rm -rf .turbo/runs
+  $ cp "$TESTDIR/fixture-configs/global_pt.json" "$(pwd)/turbo.json" && git commit --allow-empty -am "no comment" --quiet
+  $ ${TURBO} run build --summarize > /dev/null
+  $ cat .turbo/runs/*.json | jq -r '.envMode'
+  strict
+  $ cat .turbo/runs/*.json | jq -r '.tasks[0].envMode'
+  strict
+  $ cat .turbo/runs/*.json | jq -r '.tasks[0].environmentVariables | {passthrough, globalPassthrough}'
+  {
+    "passthrough": null,
+    "globalPassthrough": [
+      "GLOBAL_VAR_PT=cecd31fff1e723588eac8fe1edff89a6d2ec072f5c3bd884a98297487670b1f0"
+    ]
+  }
+
+Global passthrough specified empty array + loose
+  $ rm -rf .turbo/runs
+  $ cp "$TESTDIR/fixture-configs/global_pt-empty.json" "$(pwd)/turbo.json" && git commit --allow-empty -am "no comment" --quiet
+  $ ${TURBO} run build --experimental-env-mode=loose --summarize > /dev/null
+  $ cat .turbo/runs/*.json | jq -r '.envMode'
+  loose
+  $ cat .turbo/runs/*.json | jq -r '.tasks[0].envMode'
+  loose
+  $ cat .turbo/runs/*.json | jq -r '.tasks[0].environmentVariables | {passthrough, globalPassthrough}'
+  {
+    "passthrough": null,
+    "globalPassthrough": []
+  }
+
+Global passthrough specified value + loose
+  $ rm -rf .turbo/runs
+  $ cp "$TESTDIR/fixture-configs/global_pt.json" "$(pwd)/turbo.json" && git commit --allow-empty -am "no comment" --quiet
+  $ ${TURBO} run build --experimental-env-mode=loose --summarize > /dev/null
+  $ cat .turbo/runs/*.json | jq -r '.envMode'
+  loose
+  $ cat .turbo/runs/*.json | jq -r '.tasks[0].envMode'
+  loose
+  $ cat .turbo/runs/*.json | jq -r '.tasks[0].environmentVariables | {passthrough, globalPassthrough}'
+  {
+    "passthrough": null,
+    "globalPassthrough": [
+      "GLOBAL_VAR_PT=cecd31fff1e723588eac8fe1edff89a6d2ec072f5c3bd884a98297487670b1f0"
+    ]
```

Comment:
Fine since we're still experimental, but I was under the impression that the _value_ of pass through vars is not part of the hash. If that's still the case, the run summary could also probably omit the values.

Reply:
I think that your concern is unnecessarily conflating "is part of run summary" with "is part of hash". There are _lots_ of things in the output that aren't (intentionally) included in the hash so this doesn't strike me as a problem. We can address this in any UI: "highlight the hash inputs".

I explicitly went through to add this because being able to unmask that value in the future if you can guess the input to the hash is super-valuable to discover why the run outputs are what they are.

Reply:
> I think that your concern is unnecessarily conflating "is part of run summary" with "is part of hash"

I didn't intend it as that. My view is that the summary is a good way to understand what the tool _did_.  If it's not doing anything with a thing, it doesn't need to be reflected back to the user. In this case, my earlier comment is promoted by the fact that we do _something_ with the env var keys, but nothing with their values.

Reply:
But we _are_ doing something with it! We're passing that KV env pair into the environment! And it's going to be different on different people's boxes, but the hash will _still be the same_!

Of _all_ the places where we decide whether or not to capture this information I believe this is the _most_ important one to capture.

Reply:
That's really my distinction too: `turbo` may not care other than shuffling the correct things into the environment, but the user definitely cares what actually ended up there.

We should be doing things for the _user_.